functest/deploy_test.go

```diff
@@ -145,34 +145,32 @@ func TestDeployNamespaceMismatch(t *testing.T) {
 	sandbox, cleanup := cmdtest.TestSetupWithSandbox(t, true)
 	defer cleanup()
 
-	// first add the test repo index
-	_, err := cmdtest.AddLocalRepo(sandbox, "LocalTestRepo", filepath.Join(cmdtest.TestDirPath, "index.yaml"))
-	if err != nil {
-		t.Fatal(err)
-	}
-
 	// appsody init
 	t.Log("Running appsody init...")
-	_, err = cmdtest.RunAppsody(sandbox, "init", "nodejs-express")
+	_, err := cmdtest.RunAppsody(sandbox, "init", "nodejs-express")
 	if err != nil {
 		t.Fatal(err)
 	}
 
 	firstNamespace := "firstNamespace"
 	// appsody deploy
-	t.Log("Running appsody deploy...")
+	t.Logf("Running appsody deploy with namespace: %s ...", firstNamespace)
 	_, err = cmdtest.RunAppsody(sandbox, "deploy", "--generate-only", "-n", firstNamespace)
 	if err != nil {
 		t.Fatal(err)
 	}
 
 	secondNamespace := "secondNamespace"
 	// appsody deploy
-	t.Log("Running appsody deploy...")
+	t.Logf("Running appsody deploy with namespace: %s ...", secondNamespace)
 	_, err = cmdtest.RunAppsody(sandbox, "deploy", "--generate-only", "-n", secondNamespace)
 
-	if err != nil && !strings.Contains(err.Error(), "the namespace \""+firstNamespace+"\" from the deployment manifest does not match the namespace \""+secondNamespace+"\" passed as an argument.") {
-		t.Fatal(err)
+	if err != nil {
+		if !strings.Contains(err.Error(), "the namespace \""+firstNamespace+"\" from the deployment manifest does not match the namespace \""+secondNamespace+"\" passed as an argument.") {
+			t.Errorf("Expecting namespace error to be thrown, but another error was thrown: %s", err)
```

Comment:
I mentioned this in another PR too, but I think it is actually better to check for the string in the command output from RunAppsody rather than the err. This is because it is most important for the message to be printed to the screen for the user to read (a.k.a the output)